### PR TITLE
upgrade to CakePHP 5

### PR DESCRIPTION
**Pull request by arusinowski**
Hi, this is cakephp upload plugin updated to CakePHP 5 dev version. I will update this branch as new versions of CakePHP5 become available.

### CODE REVIEW

**Comment by ADmad** [comment]
To get the testsuite running you'll also need to update the phpunit config and the tests bootstrap to use the new fixture schema loading. You can refer to the `cake-5` of the [search](https://github.com/FriendsOfCake/search/tree/cake-5) plugin for the same.

File: src/File/Path/DefaultProcessor.php [hunk 1]
```diff
@@ -18,46 +19,46 @@ class DefaultProcessor implements ProcessorInterface
...
     /**
      * Constructor
      *
-     * @param \Cake\ORM\Table  $table the instance managing the entity
+     * @param \Cake\ORM\Table $table the instance managing the entity
      * @param \Cake\Datasource\EntityInterface $entity the entity to construct a path for.
-     * @param \Psr\Http\Message\UploadedFileInterface|string     $data the data being submitted for a save or filename stored in db
-     * @param string           $field the field for which data will be saved
-     * @param array            $settings the settings for the current field
+     * @param string|\Psr\Http\Message\UploadedFileInterface|array $data the data being submitted for a save or filename stored in db
```
Comment (ADmad) [inline src/File/Path/DefaultProcessor.php:57]:
```suggestion
     * @param string|\Psr\Http\Message\UploadedFileInterface $data the data being submitted for a save or filename stored in db
```

File: src/File/Path/DefaultProcessor.php [hunk 2]
```diff
@@ -18,46 +19,46 @@ class DefaultProcessor implements ProcessorInterface
...
+     * @param string $field the field for which data will be saved
+     * @param array $settings the settings for the current field
      */
-    public function __construct(Table $table, EntityInterface $entity, $data, string $field, array $settings)
+    public function __construct(Table $table, EntityInterface $entity, string|UploadedFileInterface|array $data, string $field, array $settings)
```
Comment (ADmad) [inline src/File/Path/DefaultProcessor.php:61]:
```suggestion
    public function __construct(Table $table, EntityInterface $entity, string|UploadedFileInterface $data, string $field, array $settings)
```

**Comment by LordSimal** [comment]
@arusinowski we recently released CakePHP 5.0 RC1 which also included a major PHPUnit update from 9 to 10.
Do you want to continue/finish this PR?

**Comment by arusinowski** [comment]
> @arusinowski we recently released CakePHP 5.0 RC1 which also included a major PHPUnit update from 9 to 10. Do you want to continue/finish this PR?

Yes, i already started working on tests upgrade

**Comment by LordSimal** [comment]
FYI: The new Cake5 CI workflow doesn't require having phpstan and psalm in the composer.json anymore.
instead all you have to do is add [this file ](https://github.com/cakephp/cakephp/blob/5.x/.phive/phars.xml) to the repo (inside the `.phive` folder of course)
Then github CI can check stan as well.

You can also adjust thecomposer.json scripts section to contain the additional commands like you can see here:
https://github.com/cakephp/debug_kit/blob/5.x/composer.json#L54-L62
Then you can check stan locally as well.

File: .phive/phars.xml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<phive xmlns="https://phar.io/phive">
+  <phar name="phpstan" version="1.10.14" installed="1.10.14" location="./tools/phpstan" copy="false"/>
+  <phar name="psalm" version="^5.13.0" installed="5.13.0" location="./tools/psalm" copy="false"/>
+  <phar name="phpunit" version="^10.1.0" installed="10.2.2" location="./tools/phpunit" copy="false"/>
```
Comment (LordSimal) [inline .phive/phars.xml:5]: PHPUnit shouldn't be part of this here

Comment (arusinowski) [reply .phive/phars.xml]: Done

**Pull request merged**